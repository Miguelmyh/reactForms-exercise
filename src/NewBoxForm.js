import React, { useState } from "react";

const NewBoxForm = ({ addBox }) => {
  const INITIAL_STATE = {
    width: "",
    height: "",
    backgroundColor: "",
  };
  const [formData, setForm] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setForm((formData) => ({ ...formData, [name]: value }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addBox(formData);
    setForm(INITIAL_STATE);
  };

  const { width, height, backgroundColor } = formData;
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="width">Box's width: </label>
      <input
        id="width"
        type="text"
        name="width"
        placeholder="Type the width for a box"
        value={width}
        onChange={handleChange}
      />
      <label htmlFor="height">Box's height: </label>
      <input
        id="height"
        type="text"
        name="height"
        placeholder="Type the height for a box"
        value={height}
        onChange={handleChange}
      />
      <label htmlFor="backgroundColor">Box's color: </label>
      <input
        id="backgroundColor"
        type="text"
        name="backgroundColor"
        placeholder="Type the color for a box"
        value={backgroundColor}
        onChange={handleChange}
      />
      <button onSubmit={handleSubmit}>Add Box</button>
    </form>
  );
};

export default NewBoxForm;
