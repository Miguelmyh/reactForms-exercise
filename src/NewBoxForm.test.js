import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

test("should render", () => {
  render(<NewBoxForm />);
});

test("should match snapshot", () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

test("shoul do some", () => {
  const { queryByPlaceholderText, queryByText, getByTestId } = render(
    <NewBoxForm />
  );
  const width = queryByPlaceholderText("Type the width for a box");
  const height = queryByPlaceholderText("Type the height for a box");
  const backgroundColor = queryByPlaceholderText("Type the color for a box");
  const btn = queryByText("Add Box");
  expect(height).toBeInTheDocument();
  expect(backgroundColor).toBeInTheDocument();
});
