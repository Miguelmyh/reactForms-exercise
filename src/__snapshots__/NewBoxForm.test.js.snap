// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Box's width: 
    </label>
    <input
      id="width"
      name="width"
      placeholder="Type the width for a box"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Box's height: 
    </label>
    <input
      id="height"
      name="height"
      placeholder="Type the height for a box"
      type="text"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Box's color: 
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Type the color for a box"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
